Detectors/Vertexing/StrangenessTracking/src/StrangenessTracker.cxx

```diff
@@ -242,58 +242,62 @@ void StrangenessTracker::processCascade(int iCasc, const Cascade& casc, const Ca
 
 void StrangenessTracker::process3Body(int i3Body, const Decay3Body& dec3body, const Decay3BodyIndex& dec3bodyIdx, int iThread)
 {
-  ClusAttachments structClus;
-  auto& daughterTracks = mDaughterTracks[iThread];
-  daughterTracks.resize(3); // resize to 3 prongs
-
-  StrangeTrack strangeTrack;
-  strangeTrack.mPartType = dataformats::kStrkThreeBody;
-  auto dec3bodyR = std::sqrt(dec3body.calcR2());
-  auto iBins3Body = mUtils.getBinRect(dec3body.getEta(), dec3body.getPhi(), mStrParams->mEtaBinSize, mStrParams->mPhiBinSize);
-  for (int& iBin3Body : iBins3Body) {
-    for (int iTrack{mTracksIdxTable[iBin3Body]}; iTrack < TMath::Min(mTracksIdxTable[iBin3Body + 1], int(mSortedITStracks.size())); iTrack++) {
-      strangeTrack.mMother = (o2::track::TrackParCovF)dec3body;
-      /// TODO: indices of daughters...
-      daughterTracks[kV0DauPos] = dec3body.getProng(kV0DauPos); // proton
-      daughterTracks[kV0DauNeg] = dec3body.getProng(kV0DauNeg); // pion
-      daughterTracks[kBach] = dec3body.getProng(kBach);         // deuteron
-      const auto& itsTrack = mSortedITStracks[iTrack];
-      const auto& ITSindexRef = mSortedITSindexes[iTrack];
-      if (mStrParams->mVertexMatching && (mITSvtxBrackets[ITSindexRef].getMin() > dec3bodyIdx.getVertexID() || mITSvtxBrackets[ITSindexRef].getMax() < dec3bodyIdx.getVertexID())) {
-        continue;
-      }
-      if (matchDecayToITStrack(dec3bodyR, strangeTrack, structClus, itsTrack, daughterTracks, iThread)) {
-        auto propInstance = o2::base::Propagator::Instance();
-        o2::track::TrackParCov decayVtxTrackClone = strangeTrack.mMother; // clone track and propagate to decay vertex
-        if (!propInstance->propagateToX(decayVtxTrackClone, strangeTrack.mDecayVtx[0], getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
-          LOG(debug) << "Mother propagation to decay vertex failed";
+  if (!mStrParams->mSkip3Body) {
```

Comment:
Why do you need to check this here if it is already check before calling this method in https://github.com/AliceO2Group/AliceO2/pull/13182/files#diff-6c56393722a05e498b450726c3c3b759d0722698379743b271c92c5f82bf8197R319-R323 ?

Reply:
HI @shahor02 
As far as I understand the `StrangenessTracker::process()` is never called from within the SVertexer but only the individual `processV0`, `processCascade` and `process3Body` functions. This is why I followed the existing logic for V0s and cascades and added the check here in the SVertexer code. The check within the `StrangenessTracker::process()` allows to still run the StrangenessTracker as standalone task calling the `StrangenessTracker::process()` function.